ui/apps/platform/src/Containers/Violations/ViolationsTableSearchFilter.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { Toolbar, ToolbarGroup, ToolbarContent, ToolbarItem } from '@patternfly/react-core';
+
+import useURLSearch from 'hooks/useURLSearch';
+import {
+    makeFilterChipDescriptors,
+    onURLSearch,
+} from 'Components/CompoundSearchFilter/utils/utils';
+import { CompoundSearchFilterConfig, OnSearchPayload } from 'Components/CompoundSearchFilter/types';
+import SearchFilterChips from 'Components/PatternFly/SearchFilterChips';
+import CompoundSearchFilter from 'Components/CompoundSearchFilter/components/CompoundSearchFilter';
+import {
+    Category as PolicyCategory,
+    Name as PolicyName,
+} from 'Components/CompoundSearchFilter/attributes/policy';
+import {
+    ID as ClusterID,
+    Name as ClusterName,
+} from 'Components/CompoundSearchFilter/attributes/cluster';
+import {
+    ID as NamespaceID,
+    Name as NamespaceName,
+} from 'Components/CompoundSearchFilter/attributes/namespace';
+import {
+    ID as DeploymentID,
+    Name as DeploymentName,
+} from 'Components/CompoundSearchFilter/attributes/deployment';
+
+const searchFilterConfig: CompoundSearchFilterConfig = [
+    {
+        displayName: 'Policy',
+        searchCategory: 'ALERTS',
+        attributes: [PolicyName, PolicyCategory],
+    },
+    {
+        displayName: 'Cluster',
+        searchCategory: 'ALERTS',
+        attributes: [ClusterID, ClusterName],
+    },
+    {
+        displayName: 'Namespace',
+        searchCategory: 'ALERTS',
+        attributes: [NamespaceID, NamespaceName],
+    },
+    {
+        displayName: 'Deployment',
+        searchCategory: 'ALERTS',
+        attributes: [DeploymentID, DeploymentName],
+    },
+];
+
+function ViolationsTableSearchFilter() {
+    const { searchFilter, setSearchFilter } = useURLSearch();
```

Comment:
Subjective/nit: Lately I've been finding that keeping the `useURL...()` helpers and other global state type hooks as close to the page level as possible and passing them down to child components feels cleaner (if a little more verbose) and prevents potential refactors down the line.

I've found in Workload CVEs that there were a handful of cases where `useURLSearch()` is called at multiple depths down the component tree, and it makes it a little difficult to see where changes are happening when debugging, as well as sometimes requiring lifting this state up in order to handle things like analytics that I didn't want to have scattered through a tree of components.

Reply:
Ah, I see. I imagined the opposite and did it this way because I thought it would be simpler. If we ever need to refactor, wouldn't it be more work if we're prop drilling and potentially affecting multiple components? At least for the URL case, since the hook directly accesses the URL state (source of truth), we could keep it self-contained in a component like this. But I haven't run into the debugging issues like you did for Workload CVEs so I may be missing something. The analytics point is valid. I'm good with keeping it higher up on the page too. If we're already doing that, we should be consistent.  I'll make adjustments 👍🏼

Reply:
Yeah it will be a little more work with prop drilling, which is the main downside I think. Here is a more concrete example of where this can be a bit of a pain to deal with:

I ran into a situation recently where `useURLSearch()` was at one point called at three different depths of the component tree. It has been reduced to only twice now:
```
WorkloadCVEOverviewPage (calls hook)
-> AdvancedFiltersToolbar (-used to- call hook)
---> SearchFilterChips (calls hook)
```
The problem here is that the callback at the top level `WorkloadCVEOverviewPage` for when the filters change also contains a call to reset pagination `pagination.setPage(1)` since a change in the filters would result in a change in the data set appearing on screen. However this `pagination` object is only accessible at the top level, so the call in the `SearchFilterChips` does _not_ reset pagination, resulting in slightly buggy/inconsistent behavior when the user deletes filters via the chip bar. (The analytics tracking isn't as big of a deal here since the `SearchFilterChips` only delete filters, which isn't an event we want to track.)

The two fixes I've thought of are either:
1. Add a call to `useURLPagination()` directly in the `SearchFilterChips`, which IMO compounds the problem by adding another global state mutation deeper in the component tree. This also mixes concerns a bit, since now the `SearchFilterChips` needs to worry about pagination, which in theory might not be relevant for every use of the component.
2. Remove the call to `useURLSearch()` in the `SearchFilterChips` component and pass in a `searchFilter` object and `onFilterChange` function. The latter could be just `setSearchFilter` directly, or some callback that wraps `setSearchFilter`, or something else. The nice part about this approach is that the child component no longer needs to care -- it just calls whatever function was provided by the parent. This can be useful in testing as well, since we could pass some function that doesn't rely on the browser's URL in order to test functionality closer to a unit level than an integration level.

Reply:
FWIW too - I don't think we follow this pattern consistently yet. I expect usage to be a mix but I tend to prefer keeping anything impure as close to the top level `Page` component as possible as it seems like over time these things work their way upwards eventually.